### ISSUE TITLE
Keyboard Shortcut

### ISSUE THREAD

**Issue by pasta12** [ryanmeisters/Xcode-Plugin-Open-Sublime-Text#9]
Instead of suggesting the user manually sets a keyboard shortcut in system preferences, the plugin could come with a keyboard shortcut.

cmd + shift + s is already used by Xcode for Save As (Duplicate), but ctrl + shift + s could work.

**Comment by ryanmeisters** [comment]
Yeah. I've looked into that a little bit. This would require us to figure out how to access Xcode's key bindings. As far as I know, I don't think any other Xcode plugins have figured out how to do that yet. I think a lot of people would be pretty excited if we did figure it out.

I've switched to control + command + s lately, which is bound to "create new snapshot" or something like that by default.

**Comment by pasta12** [comment]
I'm working on a plugin of my own and I have a key binding associated with the menu item. So it is possible. I should theoretically be able to add a keyboard shortcut if you want. It would have to be something that is already taken. I suggested ctrl + shift + s because it's not taken by anything in Xcode yet.

**Comment by ryanmeisters** [comment]
Oh nice! How do you set the key binding? Does it actually show up in the Xcode preferences?

**Comment by pasta12** [comment]
There are `setKeyEquivalentModifierMask:` and `setKeyEquivalent:` methods on `NSMenuItem`. I haven't looked into how to get it into Xcode's key bindings settings so people can customize it yet. I'm assuming that is possible as well.

**Comment by ryanmeisters** [comment]
Ah, gotcha. I kind of think that we shouldn't ship the plugin with a keyboard shortcut that can't be changed easily. It would be sweet to figure out how to make it available in the Xcode keybinding settings. A class dump might point us in some direction there. 

I guess the alternative would be to add some kind of plugin settings dialog.. but I don't love that idea either.
